delivery/services/file_system_service.py

```diff
@@ -81,16 +83,46 @@ def abspath(path):
         """
         return os.path.abspath(path)
 
+    def create_parent_dirs(self, child_path):
+        """
+        Create the parent directory structure for a child path
+        :param child_path: path to child
+        :return: None
+        """
+        self.makedirs(self.dirname(child_path), exist_ok=True)
+
     def symlink(self, source, link_name):
         """
         Shadows os.symlink
         :param source: of link
         :param link_name: the name of the link to create
         :return: None
         """
-        self.makedirs(self.dirname(link_name), exist_ok=True)
+        self.create_parent_dirs(link_name)
+        # replace with pathlib.Path(link_name).symlink_to(source) ?
```

Comment:
I see your point although I'm not too fuzzed about the extra `FileSystemService` layer tbh, it's easy to understand what it does and gives a bit of control.

That said, I'm not particularly attached to it either, so I don't mind if it's removed. As you say though, I think it would take some work so I agree that it doesn't belong in this PR.